crates/common/consensus/src/deneb/beacon_state.rs

```diff
@@ -1282,6 +1284,50 @@ impl BeaconState {
         }
         Ok(())
     }
+
+    pub fn process_eth1_data_reset(&mut self) -> anyhow::Result<()> {
+        let next_epoch = self.get_current_epoch() + 1;
+
+        // Reset eth1 data votes
+        if next_epoch % EPOCHS_PER_ETH1_VOTING_PERIOD == 0 {
+            self.eth1_data_votes = VariableList::default();
+        }
+
+        Ok(())
+    }
+
+    pub fn process_effective_balance_updates(&mut self) -> anyhow::Result<()> {
+        // Update effective balances with hysteresis
+        for (index, validator) in self.validators.iter_mut().enumerate() {
+            let balance = self.balances[index];
+            let hysteresis_increment = EFFECTIVE_BALANCE_INCREMENT / HYSTERESIS_QUOTIENT;
+            let downward_threshold = hysteresis_increment * HYSTERESIS_DOWNWARD_MULTIPLIER;
+            let upward_threshold = hysteresis_increment * HYSTERESIS_UPWARD_MULTIPLIER;
+
+            if balance + downward_threshold < validator.effective_balance
+                || validator.effective_balance + upward_threshold < balance
+            {
+                validator.effective_balance =
+                    (balance - balance % EFFECTIVE_BALANCE_INCREMENT).min(MAX_EFFECTIVE_BALANCE);
+            }
+        }
+        Ok(())
+    }
+
+    pub fn process_historical_roots_update(&mut self) -> anyhow::Result<()> {
+        // Set historical root accumulator
+        let next_epoch = self.get_current_epoch() + 1;
+        if next_epoch % (SLOTS_PER_HISTORICAL_ROOT / SLOTS_PER_EPOCH) == 0 {
+            let historical_batch = HistoricalBatch {
+                block_roots: self.block_roots.clone(),
+                state_roots: self.state_roots.clone(),
+            };
+            self.historical_roots
+                .push(historical_batch.tree_hash_root())
+                .map_err(|err| anyhow!("Failed to get historical_roots {err:?}"))?;
+        }
+        Ok(())
+    }
```

Comment:
This function isn't in deneb it was replaced by `process_historical_summaries_update` in capella